Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -313,9 +324,15 @@ internal static string DeterminePrimaryScriptFile(JObject functionConfig, string
                         Path.GetFileNameWithoutExtension(p).ToLowerInvariant() == "run" ||
                         Path.GetFileName(p).ToLowerInvariant() == "index.js");
                 }
+            }
 
-                return functionPrimary;
+            if (string.IsNullOrEmpty(functionPrimary))
+            {
+                throw new ConfigurationErrorsException("Unable to determine the primary function script. Try renaming your entry point script to 'run' (or 'index' in the case of Node), " +
+                    "or alternatively you can specify the name of the entry point script explicitly by adding a 'scriptFile' property to your function metadata.");
```

Comment:
I wonder if the better behavior would be to just have a null `ScriptHref` if we can't figure out the primary script. This way, they still show up in the portal. Talk to @alrod to make sure Portal can handle that. I think generally we want to be fairly lenient so that user can still open function in portal and fix issue. If you skip it, it just disappears.